src/forms/renderers/OAuth/useOauthHandler.ts

```diff
@@ -0,0 +1,97 @@
+import { accessToken, authURL } from 'api/oauth';
+import { useOAuth2 } from 'hooks/forks/react-use-oauth2/components';
+import { isEmpty } from 'lodash';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useDetailsForm_connectorImage_connectorId } from 'stores/DetailsForm/hooks';
+import { useEndpointConfigStore_endpointConfig_data } from 'stores/EndpointConfig/hooks';
+import { CREDENTIALS, INJECTED_VALUES } from './shared';
+
+// Hook for OAuth popup opening, error handling, error message setting, etc.
+export const useOauthHandler = (
+    provider: string | undefined,
+    successHandler: (tokenResponse: any) => void
+) => {
+    const intl = useIntl();
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const endpointConfigData = useEndpointConfigStore_endpointConfig_data();
+    const imageTag = useDetailsForm_connectorImage_connectorId();
+
+    // handler for the useOauth stuff
+    const onError = (error: any) => {
+        if (error === 'access_denied') {
+            setErrorMessage(
+                intl.formatMessage(
+                    {
+                        id: 'oauth.authentication.denied',
+                    },
+                    { provider }
+                )
+            );
+        } else {
+            setErrorMessage(error);
+        }
+    };
+
+    // handler for the useOauth stuff
+    const onSuccess = async (payload: any) => {
+        const preparedData = endpointConfigData;
+        preparedData[CREDENTIALS] = {
+            ...endpointConfigData[CREDENTIALS],
+            ...INJECTED_VALUES,
+        };
```

Comment:
This is the only real change to this logic... we not make sure the injected value is here. 

Probably does not matter as I don't think the accessToken endpoint uses them but wanted to be careful.